common/src/main/java/com/aliyun/rss/common/network/server/TransportRequestHandler.java

```diff
@@ -0,0 +1,290 @@
+  }
+
+  @Override
+  public void channelActive() {
+    rpcHandler.channelActive(reverseClient);
+  }
+
+  @Override
+  public void channelInactive() {
+    if (streamManager != null) {
+      try {
+        streamManager.connectionTerminated(channel);
+      } catch (RuntimeException e) {
+        logger.error("StreamManager connectionTerminated() callback failed.", e);
+      }
+    }
+    rpcHandler.channelInactive(reverseClient);
+  }
+
+  @Override
+  public void handle(RequestMessage request) {
+    if (request instanceof ChunkFetchRequest) {
+      if (checkRegistered(request)) {
+        processFetchRequest((ChunkFetchRequest) request);
+      }
+    } else if (request instanceof RpcRequest) {
+      if (checkRegistered(request)) {
+        processRpcRequest((RpcRequest) request);
+      }
+    } else if (request instanceof OneWayMessage) {
+      if (checkRegistered(request)) {
+        processOneWayMessage((OneWayMessage) request);
+      }
+    } else if (request instanceof PushData) {
+      if (checkRegistered(request)) {
+        processPushData((PushData) request);
+      }
+    } else if (request instanceof PushMergedData) {
+      if (checkRegistered(request)) {
+        processPushMergedData((PushMergedData) request);
+      }
+    } else {
+      throw new IllegalArgumentException("Unknown request type: " + request);
+    }
+  }
+
+  private boolean checkRegistered(RequestMessage req) {
+    if (!rpcHandler.checkRegistered()) {
+      IOException e = new IOException("Worker Not Registered!");
+      if (req instanceof RpcRequest) {
+        respond(new RpcFailure(((RpcRequest)req).requestId, Throwables.getStackTraceAsString(e)));
+      } else if (req instanceof ChunkFetchRequest) {
+        respond(new ChunkFetchFailure(((ChunkFetchRequest)req).streamChunkId,
+            Throwables.getStackTraceAsString(e)));
+      } else if (req instanceof OneWayMessage) {
+        logger.warn("Ignore OneWayMessage since worker is not registered!");
+      }
+      return false;
+    }
+    return true;
+  }
+
+  private void processFetchRequest(final ChunkFetchRequest req) {
+    if (source != null) {
+      source.startTimer(NetWorkSource.FetchChunkTime(), req.toString());
+    }
+    if (logger.isTraceEnabled()) {
+      logger.trace("Received req from {} to fetch block {}", NettyUtils.getRemoteAddress(channel),
+        req.streamChunkId);
+    }
+    long chunksBeingTransferred = streamManager.chunksBeingTransferred();
+    if (chunksBeingTransferred >= maxChunksBeingTransferred) {
+      logger.warn("The number of chunks being transferred {} is above {}, close the connection.",
+        chunksBeingTransferred, maxChunksBeingTransferred);
+      channel.close();
+      if (source != null) {
+        source.stopTimer(NetWorkSource.FetchChunkTime(), req.toString());
+      }
+      return;
+    }
+    ManagedBuffer buf;
+    try {
+      streamManager.checkAuthorization(reverseClient, req.streamChunkId.streamId);
```

Comment:
What's the difference between rpc and stream manager, why we not unify the request processing, we can divide the rpc processing into two parts, one for reading and one for writing, both are based on the shared rpc layer.